rest/blip_sync.go

```diff
@@ -71,3 +79,31 @@ func (h *handler) handleBLIPSync() error {
 
 	return nil
 }
+
+func (sc *ServerContext) limitConcurrentReplications(ctx context.Context) (func(sc *ServerContext), error) {
+	if sc.Config.Replicator.MaxConcurrentConnections == 0 {
+		return nil, nil
+	}
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	capacity := sc.Config.Replicator.MaxConcurrentConnections
```

Comment:
Is the reason to lock the whole config here rather than the atomic because we care about config value and also the counter value?